recipes/libmp3lame/all/conanfile.py

```diff
@@ -10,70 +13,90 @@ class LibMP3LameConan(ConanFile):
-    _autotools = None
 
-    @property
-    def _use_winbash(self):
-        return tools.os_info.is_windows and (self.settings.compiler == "gcc" or tools.cross_building(self.settings))
+    exports_sources = ["patches/**"]
+    _autotools = None
 
     @property
     def _is_msvc(self):
-        return self.settings.compiler == 'Visual Studio'
+        return self.settings.compiler == "Visual Studio"
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
             del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if not self._is_msvc:
+            self.build_requires("gnu-config/cci.20201022")
+            if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+                self.build_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "lame-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def _apply_patch(self):
-        tools.replace_in_file(os.path.join(self._source_subfolder, 'include', 'libmp3lame.sym'), 'lame_init_old\n', '')
-        for patch in [6410, 6416]:
-            tools.patch(base_path=self._source_subfolder, patch_file='%s.patch' % patch, strip=3)
-        tools.patch(base_path=self._source_subfolder, patch_file='android.patch')
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "include", "libmp3lame.sym"), "lame_init_old\n", "")
 
-    def _build_vs(self):
+    @contextmanager
+    def _msvc_build_environment(self):
         with tools.chdir(self._source_subfolder):
-            shutil.copy('configMS.h', 'config.h')
-            command = 'nmake -f Makefile.MSVC comp=msvc asm=yes'
-            if self.settings.arch == 'x86_64':
-                tools.replace_in_file('Makefile.MSVC', 'MACHINE = /machine:I386', 'MACHINE =/machine:X64')
-                command += ' MSVCVER=Win64'
-            if self.options.shared:
-                command += ' dll'
-            with tools.vcvars(self.settings, filter_known_paths=False, force=True):
-                self.run(command)
+            with tools.vcvars(self.settings):
+                with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
+                    yield
+
+    def _build_vs(self):
+        with self._msvc_build_environment():
+            shutil.copy("configMS.h", "config.h")
```

Comment:
did not remove this line, but it's useless I think, `Makefile.MSVC` should take care of that.